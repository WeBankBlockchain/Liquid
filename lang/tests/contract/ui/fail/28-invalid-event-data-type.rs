use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(event)]
    struct TestEvent {
        f: f32,
    }

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) -> () {
            self.env().emit(TestEvent { f: 1.0 });
        }
    }
}

fn main() {}
