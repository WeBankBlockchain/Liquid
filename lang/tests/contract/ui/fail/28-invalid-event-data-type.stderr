error[E0277]: the trait bound `f32: You_Should_Use_An_Valid_Output_Type` is not satisfied
  --> $DIR/28-invalid-event-data-type.rs:10:12
   |
10 |         f: f32,
   |            ^^^ the trait `You_Should_Use_An_Valid_Output_Type` is not implemented for `f32`
